server/src/services/hcs-message-listener.service.ts

```diff
@@ -180,20 +180,26 @@ export class HcsMessageListenerService {
 	private async computeStreamStartingTimestamp() {
 		const latestMessage = await this.mirror.waitForHcsMessageInfo(this.sequenceNumber);
 		const timestamp = keyString_to_timestamp(latestMessage.consensus_timestamp as TimestampKeyString);
+		const envTimestamp = keyString_to_timestamp(this.config.hcsStartDate);
+
+		if (this.sequenceNumber === 1 && (!envTimestamp || (envTimestamp.seconds === 0 && envTimestamp.nanos === 0))) {
```

Comment:
This update just makes sure that we read in sequence 1 off the topic which we didn't have to do before because the original rule was manually read off the topic instead in another function.

The default envTimestamp on init is "0.0" which is why we need that extra check.

The user could still choose to start the sequence from a later timestamp. The user would just need to make sure they start from the timestamp just before an existing rule as rule one is no longer read in by default.